tagstudio/src/core/library/alchemy/fields.py

```diff
@@ -0,0 +1,173 @@
+
+    entry_id: Mapped[int] = mapped_column(ForeignKey("entries.id"))
+    entry: Mapped[Entry] = relationship(foreign_keys=[entry_id])
+
+    tags: Mapped[set[Tag]] = relationship(secondary="tag_fields")
+
+    def __key(self):
+        return (
+            self.entry_id,
+            self.type_key,
+        )
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, value) -> bool:
+        if isinstance(value, TagBoxField):
+            return self.__key() == value.__key()
+        raise NotImplementedError
+
+
+class DatetimeField(Base):
+    __tablename__ = "datetime_fields"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    type_key: Mapped[str] = mapped_column(ForeignKey("library_fields.key"))
+    type: Mapped[LibraryField] = relationship(foreign_keys=[type_key], lazy=False)
+
+    entry_id: Mapped[int] = mapped_column(ForeignKey("entries.id"))
+    entry: Mapped[Entry] = relationship(foreign_keys=[entry_id])
+
+    value: Mapped[datetime.datetime | None]
+
+    def __key(self):
+        return (self.type, self.value)
+
+    def __hash__(self):
+        return hash(self.__key())
+
+    def __eq__(self, value) -> bool:
+        if isinstance(value, DatetimeField):
+            return self.__key() == value.__key()
+        raise NotImplementedError
+
+
+@dataclass
+class DefaultField:
+    id: int
+    name: str
+    type: Any  # TextFieldTypes | TagBoxTypes | DateTimeTypes
+
+
+class _FieldID(Enum):
+    """Only for bootstrapping content of DB table"""
+
+    TITLE = DefaultField(id=0, name="Title", type=FieldTypeEnum.TEXT_LINE)
+    AUTHOR = DefaultField(id=1, name="Author", type=FieldTypeEnum.TEXT_LINE)
+    ARTIST = DefaultField(id=2, name="Artist", type=FieldTypeEnum.TEXT_LINE)
+    URL = DefaultField(id=3, name="URL", type=FieldTypeEnum.TEXT_LINE)
+    DESCRIPTION = DefaultField(id=4, name="Description", type=FieldTypeEnum.TEXT_LINE)
+    NOTES = DefaultField(id=5, name="Notes", type=FieldTypeEnum.TEXT_BOX)
+    TAGS = DefaultField(id=6, name="Tags", type=FieldTypeEnum.TAGS)
+    TAGS_CONTENT = DefaultField(id=7, name="Content Tags", type=FieldTypeEnum.TAGS)
+    TAGS_META = DefaultField(id=8, name="Meta Tags", type=FieldTypeEnum.TAGS)
+    COLLATION = DefaultField(id=9, name="Collation", type=FieldTypeEnum.TEXT_LINE)
+    DATE = DefaultField(id=10, name="Date", type=FieldTypeEnum.DATETIME)
+    DATE_CREATED = DefaultField(id=11, name="Date Created", type=FieldTypeEnum.DATETIME)
+    DATE_MODIFIED = DefaultField(
+        id=12, name="Date Modified", type=FieldTypeEnum.DATETIME
+    )
+    DATE_TAKEN = DefaultField(id=13, name="Date Taken", type=FieldTypeEnum.DATETIME)
+    DATE_PUBLISHED = DefaultField(
+        id=14, name="Date Published", type=FieldTypeEnum.DATETIME
+    )
+    # ARCHIVED = DefaultField(id=15, name="Archived",  type=CheckboxField.checkbox)
+    # FAVORITE = DefaultField(id=16, name="Favorite", type=CheckboxField.checkbox)
```

Comment:
Are this not needed?

Reply:
I didnt find it used anywhere; it was probably used in past before `Favorite` and `Archived` was changed to a tag, but I want to keep it around just in case

Reply:
fair!